src/Employer/Employer.Web/Orchestrators/EntityValidatingOrchestrator.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Esfa.Recruit.Vacancies.Client.Application.Validation;
+
+namespace Esfa.Recruit.Employer.Web.Orchestrators
+{
+    public abstract class EntityValidatingOrchestrator<TEntity, TViewModel>
+    {
+        private readonly EntityToViewModelPropertyMappings<TEntity, TViewModel> _mappings;
+        private readonly IDictionary<string, string> _mappingDictionary;
+
+        public EntityValidatingOrchestrator()
+        {
+            _mappings = DefineMappings();
+            
+            _mappingDictionary = BuildMappingDictionary();
```

Comment:
Some thoughts in the shower this morning...  

Once we've populated the `_mappingDictionary` I don't think we care about `_mappings`. Maybe this can just be a local variable and not private?

The mapping setup happens on every request although the mapping functionality doesn't change between requests or sessions. So I'm wondering if the mappings could be a singleton and generated at app start - Just a thought for a possible future improvident, obs not needed for now.

Reply:
Yeah I had the same thought about the repeated generation of the mappings. Will take a look at it.